Support for Bionic and pre-1.0 versions

CloudSurgeon opened the pull request:
1. I updated some deprecated code.
2. Created two vars files for xenial and bionic to account for the different packages
3. The mysql user is created for both localhost & 127.0.0.1 access
4. Added a gcc_version var (gcc-6 by default) to account for [GUACAMOLE-500]
5. Added support for pre-1.0.0 guacamole versions
6. Added fix for issue seen in debuerreotype/debuerreotype#10

Commit:
add the user for 127.0.0.1, as well as localhost

Commit:
added compiler selection

Commit:
split variables out by major distribution

Commit:
loop via squash_action is deprecated

Commit:
supports bionic

Commit:
corrected name

Commit:
allows for archived (pre-1.0.0) packages

Commit:
Fix for https://github.com/debuerreotype/debuerreotype/issues/10

Commit:
added ability to specify application url

alexfeig commented:
Hey @CloudSurgeon - thanks for making these changes. I haven't had a chance to test them yet, but wanted to let you know I'm appreciative of your work and it looks good! Hopefully have time to test this soon :)

CloudSurgeon commented:
No problem. I tested them via Molecule. If you like, I can file a pull request to add my molecule tests, too.

> On Jan 31, 2019, at 1:48 PM, Alex Feigenson <notifications@github.com> wrote:
> 
> Hey @CloudSurgeon - thanks for making these changes. I haven't had a chance to test them yet, but wanted to let you know I'm appreciative of your work and it looks good! Hopefully have time to test this soon :)
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

alexfeig commented:
Wow - that'd be great! I had always imagined putting this into Travis, but never got around to it.

CloudSurgeon commented:
Yep. that would be the next step. You can just point Travis to the yaml and go. I’ll do another pull this weekend

> On Jan 31, 2019, at 2:17 PM, Alex Feigenson <notifications@github.com> wrote:
> 
> Wow - that'd be great! I had always imagined putting this into Travis, but never got around to it.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.

Commit:
added retries and molecule tests

CloudSurgeon commented:
Added the molecule tests and I also added retries to all the get_urls. I rely on an external database role as the database tasks included in this role did not offer satisfactory configuration options.

Commit:
allow for building from source
